src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const str = sourceString;
+  const result = {};
+  const items = str.split(';').filter((item) => item.trim());
+
+  items.forEach((item) => {
+    const parts = item.split(':');
+    const key = parts[0].trim();
+    const value = parts.slice(1).join(':').trim();
+
+    result[key] = value;
```

Comment:
Consider adding a check to ensure that `key` is not an empty string before assigning it to the `result` object. This will prevent empty keys from being added to the object, which could lead to unexpected behavior.